infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java

```diff
@@ -104,13 +114,47 @@
   }
 
   /**
-   * Creates specified pod.
+   * Starts the specified Pod via a Deployment.
    *
-   * @param pod pod to create
+   * @param pod pod to deploy
    * @return created pod
    * @throws InfrastructureException when any exception occurs
    */
   public Pod create(Pod pod) throws InfrastructureException {
+    putLabel(pod, CHE_WORKSPACE_ID_LABEL, workspaceId);
+    ObjectMeta metadata = pod.getMetadata();
+    PodSpec podSpec = pod.getSpec();
+    podSpec.setRestartPolicy("Always"); // Only allowable value
+    try {
+      Deployment deployment =
+          clientFactory
+              .create(workspaceId)
+              .extensions()
+              .deployments()
+              .createNew()
+              .withMetadata(metadata)
+              .withNewSpec()
+              .withReplicas(1)
+              .withNewTemplate()
+              .withMetadata(metadata)
+              .withSpec(podSpec)
+              .endTemplate()
+              .endSpec()
+              .done();
+      return awaitAndGetPod(deployment);
+    } catch (KubernetesClientException e) {
+      throw new KubernetesInfrastructureException(e);
+    }
+  }
+
+  /**
+   * Create a terminating pod that is not part of a Deployment.
+   *
+   * @param pod the Pod to create
+   * @return the created pod
+   * @throws InfrastructureException when any error occurs
+   */
+  public Pod createTemporary(Pod pod) throws InfrastructureException {
```

Comment:
Maybe we should use some kubernetes native vocabulary not to confuse with a new concept of a temporary pod?

Reply:
I agree, the naming in this class has become quite confusing. There's now somewhat of a split role in KubernetesDeployments, as its becoming more of a utility class for managing Pod resources (both through bare pods and deployments) than a simple wrapper around Pods. 

In picking this name (which I agree is not very good), I was trying to choose something that would at least not add to the confusion -- calling it `createJob` is deceptive as it currently doesn't create jobs, and calling it `createPod` is confusing since both create methods result in the creation of pods. I'm starting to lean towards finding a sensible non-kubernetes vocabulary way of describing what this class does, and renaming it away from `KubernetesDeployments`

Perhaps
```
KubernetesDeployments -> KubernetesComputeResources
    .create() -> .createNotTerminating()
    .createTemporary() -> .createTerminating()
```
This would at least bring it in line with the language used in describing [resource quotas](https://kubernetes.io/docs/concepts/policy/resource-quotas/). WDYT?

Reply:
I would say that will be created base pod terminating or not depends only on infrastructure configuration.

Maybe `#deploy(Pod)` and `create(Pod)` will make more clear that one method creates pod directly while another one wraps pod into Deployment. Then we have to document which limitations have creation bare pod.

Another way to make the clear purpose of `create(Pod)` method may be using something like that `create(Pod pod, String[] command)`. So it would be clear that this method should be used only if a user wants to execute a one-time task. We would even be able to move `PVCSubPathHelper#execute(String, String[], String...)` to KubernetesDeployment, so not only PVCSubPathHelper would be able to easily execute a command (submit task).
Guys. WHYT? Does it make sense?